lib/modules/platform/space/index.ts

```diff
@@ -0,0 +1,304 @@
+import { logger } from '../../../logger';
+import type { BranchStatus } from '../../../types';
+import { parseJson } from '../../../util/common';
+import * as git from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { trimTrailingSlash } from '../../../util/url';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfigByContent,
+  EnsureCommentRemovalConfigByTopic,
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { repoFingerprint } from '../util';
+import { SpaceClient } from './client';
+import { SpaceDao } from './dao';
+import { getSpaceRepoUrl } from './utils';
+
+export const id = 'space';
+
+const globalConfig: {
+  endpoint?: string;
+} = {};
+
+// TODO: make const?
+let repoConfig: {
+  projectKey?: string;
+  repository?: string;
+  head?: string;
+} = {};
+
+export function writeToConfig(newConfig: typeof repoConfig): void {
+  repoConfig = { ...repoConfig, ...newConfig };
+}
+
+let dao: SpaceDao;
+
+export function initPlatform({
+  endpoint,
+  token,
+}: PlatformParams): Promise<PlatformResult> {
+  logger.debug(`SPACE initPlatform(${endpoint!})`);
+
+  if (!endpoint) {
+    throw new Error('Init: You must configure your JetBrains Space endpoint');
+  }
+
+  if (!token) {
+    throw new Error('Init: You must configure a JetBrains Space token');
+  }
+
+  globalConfig.endpoint = trimTrailingSlash(endpoint);
+  dao = new SpaceDao(new SpaceClient(`https://${globalConfig.endpoint}`));
+
+  return Promise.resolve({
+    endpoint: globalConfig.endpoint,
+  });
+}
+
+export async function getRepos(): Promise<string[]> {
+  logger.debug(`SPACE getRepos()`);
+  return await dao.findRepositories();
+}
+
+export async function initRepo({
+  repository,
+}: RepoParams): Promise<RepoResult> {
+  logger.debug(`SPACE initRepo(${repository})`);
+
+  const repoParts = repository.split('/');
+  const projectKey = repoParts[0];
+  const shortRepository = repoParts[1];
+
+  repoConfig = {
+    ...repoConfig,
+    projectKey,
+    repository: shortRepository,
+  };
+
+  const baseUrl = globalConfig.endpoint!;
+  const url = getSpaceRepoUrl(repository, baseUrl);
+  await git.initRepo({ url });
+
+  return {
+    defaultBranch: await dao.findDefaultBranch(projectKey, shortRepository),
+    isFork: false,
+    repoFingerprint: repoFingerprint(repository, baseUrl),
+  };
+}
+
+export async function findPr(
+  findPRConfig: FindPRConfig,
+  refreshCache?: boolean,
+): Promise<Pr | null> {
+  logger.debug(
+    `SPACE findPr(${JSON.stringify(findPRConfig)}, ${refreshCache})`,
+  );
```

Comment:
Just a general note that logging like this will need to be lowered or removed before merge

Reply:
Right, I'll clean up most of the debug logs, thanks

Reply:
Replaced most of logs with trace or simply removed, kept a couple that might be useful for general user debugging